Adds KafkaAvro option to fail when schema missing

bfncs opened the pull request:
This is a follow up to #63 which was reverted by #65.

If there is no schema found on the schema registry for a topic's key/value, the producer currently falls back to logging a warning and producing a message as JSON. I think this behaviour is misleading, because I would assume calling Producer.produce() without causing an error to mean the message has been successfully serialized and sent which is not the case.

In this PR I added an optional options property `shouldFailWhenSchemaIsMissing` to the KafkaAvro() constructor. If this is set to true, producing a message will throw an error if no schema can be found for key or value.

I would still prefer to have this option set to true by default and require users to explicitly opt out of the current, misleading behavior, this would require a new major release, though - would you consider adding this for 2.x?

ricardohbin commented on package.json:
```diff
@@ -13,6 +13,10 @@
+    },
+    {
+      "name": "Marc Löhe",
+      "email": "marcloehe@gmail.com"
```
Please, can you add yourself in Release History in README (that would be `1.2.1`) instead of here? Because we use this field to main contribuitors only

bfncs commented:
Sorry, I completely forgot about this PR. I added the parameter to the README and the contribution history and removed myself from the contributors in `package.json` - hope this is ok now?

ricardohbin commented:
@bfncs perfect! I will bump it before I merge this major (https://github.com/waldophotos/kafka-avro/pull/70)
